### DISCUSSION TITLE
Refusing to load URL as it exceeds 2097152 characters

### DISCUSSION THREAD

**Discussion by Djutza** [cefsharp/CefSharp#4165] [General]
Hi,

it seems that LoadHtml cannot be used for HTML bigger than 2MB. It doesn't show anything on screen. I assume it is a limitation in chromium itself. 
I know one workaround would be to save HTML text in temporary file and then use LoadUrl with file path. Sincerely, I am not a fan on using temporary files.
Any other workaround?

Cheers!

**Comment by amaitland** [comment]
>  I assume it is a limitation in chromium itself.

It's my understanding that `Chromium` is  limited to 2mb for a url.

There are multiple overloads for `LoadHtml`, use [LoadHtml(IWebBrowser, String, String)](https://cefsharp.github.io/api/102.0.x/html/M_CefSharp_WebBrowserExtensions_LoadHtml_2.htm) instead. (provide a url for the page to display).

See https://github.com/cefsharp/CefSharp/wiki/General-Usage#loading-htmlcssjavascriptetc-from-diskdatabaseembedded-resourcestream

Alternatively implement your own [SchemeHandlerFactory](https://github.com/cefsharp/CefSharp/wiki/General-Usage#scheme-handler)

**Reply by Djutza** [reply]
Thanks @amaitland, it works like a charm.
